apps/homescreen/js/app.js

```diff
@@ -321,6 +324,21 @@ const SETTINGS_VERSION = 0;
   }
 
   App.prototype = {
+    get iconSize() {
+      // If this._iconSize is undefined or 0, let's refresh the value.
+      if (!this._iconSize) {
+        for (var i = 0; i < this.icons.childNodes.length; i++) {
+          var container = this.icons.childNodes[i].firstChild;
+          if (container.style.display !== 'none' && container.firstChild) {
```

Comment:
Do we ever add empty containers to the container? Just wondering why we have the container.firstChild check.